OpenRA.Game/Settings.cs

```diff
@@ -392,11 +399,20 @@ public Settings(string file, Arguments args)
 
 		public void Save()
 		{
-			var root = new List<MiniYamlNode>();
 			foreach (var kv in Sections)
-				root.Add(new MiniYamlNode(kv.Key, FieldSaver.SaveDifferences(kv.Value, Activator.CreateInstance(kv.Value.GetType()))));
+			{
+				var sectionYaml = yamlCache.FirstOrDefault(x => x.Key == kv.Key);
+				if (sectionYaml == null)
+				{
+					sectionYaml = new MiniYamlNode(kv.Key, new MiniYaml(""));
+					yamlCache.Add(sectionYaml);
+				}
+
+				var currentSection = FieldSaver.SaveDifferences(kv.Value, Activator.CreateInstance(kv.Value.GetType()));
+				sectionYaml.Value.Nodes = MiniYaml.Merge(new[] { sectionYaml.Value.Nodes, currentSection.Nodes });
```

Comment:
If `sectionYaml` contains a custom value and the player changes something back to its default then `currentSection` won't have an override, and the new value won't overwrite the old.